kedro/logging.py

```diff
@@ -54,3 +57,20 @@ def __init__(self, *args: Any, **kwargs: Any):
             # fixed on their side at some point, but until then we disable it.
             # See https://github.com/Textualize/rich/issues/2455
             rich.traceback.install(**traceback_install_kwargs)  # type: ignore[arg-type]
+
+    def emit(self, record: LogRecord):
```

Comment:
The trick here is to keep the raw message without markup language. The handler will be responsible to add the [brackets] (I believe this is how `rich.RichHandler` functions anyway)